pkgs/desktops/gnome-3/apps/gnome-network-displays/default.nix

```diff
@@ -0,0 +1,73 @@
+{ stdenv
+, fetchurl
+# native
+, meson
+, ninja
+, cmake
+, pkg-config
+, gettext
+, desktop-file-utils
+, appstream-glib
+, wrapGAppsHook
+, python3
+# Not native
+, gst_all_1
+, gsettings-desktop-schemas
+, gtk3
+, glib
+, networkmanager
+, libpulseaudio
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gnome-network-displays";
+  version = "0.90.4";
+
+  src = fetchurl {
+    url = "mirror://gnome/sources/${pname}/${stdenv.lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
+    sha256 = "04snnfz5jxxpjkrwa7dchc2h4shszi8mq9g3ihvsvipgzjw3d498";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    cmake
+    pkg-config
+    gettext
+    desktop-file-utils
+    appstream-glib
+    wrapGAppsHook
+    python3
+    # Needed to compile schemas in postinstall.py 
+    glib
+  ];
+
+  preConfigure = ''
+    patchShebangs ./build-aux/meson/postinstall.py
+  '';
+
+  buildInputs = [
+    gtk3
+    gsettings-desktop-schemas
+    gst_all_1.gstreamer
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    gst_all_1.gst-plugins-bad
+    gst_all_1.gst-plugins-ugly
+    gst_all_1.gst-rtsp-server
+    networkmanager
+    libpulseaudio
+  ];
+
+  # It needs this but it doesn't search for it during meson's config and without
+  # this the build fails
+  NIX_CFLAGS_COMPILE = "-I${glib.dev}/include/gio-unix-2.0";
```

Comment:
https://github.com/NixOS/nixpkgs/issues/36468

Reply:
Good to know. Is the workaround applied OK? I've mentioned the issue in the comment above.

Reply:
Yeah, the workaround is fine, but it would be nicer to open issue upstream so we can eventually get rid of it.

Reply:
The fix is easy enough and I suggested it upstream at https://gitlab.gnome.org/GNOME/gnome-network-displays/-/merge_requests/147 . I used that merge request patch applied instead of the `-I` workaround, with a link to both downstream and upstream issues.